File: js/src/backtracking/combinationSum.js
const combinationSum = (candidates, target) => {
  const result = [];
  const backtracking = (index, currentSum, sum) => {
    if (sum === target) {
      result.push([...currentSum]);
      return;
    }
    if (sum > target || index >= candidates.length) {
      return;
    }
    backtracking(
      index,
      [...currentSum, candidates[index]],
      sum + candidates[index],
    );
    backtracking(index + 1, currentSum, sum);
  };

  backtracking(0, [], 0);
  return result;
};

File: python/backtracking/combinationSum.py
class Solution:
    def combinationSum(self,candidates,target):
        result = []
        def backtracking(index,curSum, sum):
            if sum == target:
                result.append(curSum.copy())
                return
            if sum > target or index >= len(candidates):
                return

            curSum.append(candidates[index])
            backtracking(index,curSum, sum +candidates[index])
            curSum.pop()
            backtracking(index+1,curSum, sum)

        backtracking(0,[],0)
        return result
